src/realm.h

```diff
@@ -3023,7 +3023,10 @@ RLM_API void realm_app_config_set_metadata_mode(realm_app_config_t*,
 RLM_API void realm_app_config_set_metadata_encryption_key(realm_app_config_t*, const uint8_t[64]) RLM_API_NOEXCEPT;
 RLM_API void realm_app_config_set_security_access_group(realm_app_config_t*, const char*) RLM_API_NOEXCEPT;
 
-RLM_API realm_sync_client_config_t* realm_app_config_get_sync_client_config(realm_app_config_t*) RLM_API_NOEXCEPT;
```

Comment:
I don't fully understand why this getter couldn't just be implemented. The issue I see is that with only a setter available, it becomes impossible to just set one property of the sync_client_config because `realm_app_config_set_sync_client_config` will overwrite everything. This means that this API will only allow a use like this: 
1. make a new sync_client_config
2. set all the properties
3. use `realm_app_config_set_sync_client_config` to apply the changes
4. any further changes to the sync_client_config instance are not actually going to change the config that the app uses, and you'd have to do step 3 again

Whereas if we support a getter that returns a pointer to the already existing instance `AppConfig::sync_client_config` then users of this API can just get that pointer and modify it directly with the various setters, and never need to allocate memory on their side of things.

Reply:
The problem is that in the non-appservices scenario users allocate and free realm_sync_client_config_t, which requires that it inherit from WrapC. Owning and non-owning pointers need to be different types with how we've structured the C API. We could add a second type, but a simpler variation on that would be to add getters and setters for the SyncConfig fields that just take an AppConfig.

Reply:
I have been trying to add a second type that also uses the existing functions, but I don't think I've been able to get it just right without having to add separate functions for the second type. I tried creating a `realm_sync_client_config` that inherits from realm::SyncClientConfig and used by all the functions; and a `realm_sync_client_config_wrapc` that inherits from `WrapC` and `realm_sync_client_config`, which is returned by the sync_client_config_new() function....
I would be happy to just add setters for the SyncConfig fields (we usually don't have getters on most of the struct fields).

Reply:
I see, that is an unfortunate conflict that muddies the API. Whatever seems easiest is fine.
If the SDKs use the `realm_app_config_set_sync_client_config` function as it is now, then I suppose they have created a new config instance that they manage the lifetime of anyways, which means that if they want to update a single property, they could set it on their instance, and then update the entire thing again using the setter. So not the end of the world.

Reply:
I left the `realm_app_config_set_sync_client_config()` function in the code and also added individual C_API setter functions for the SyncClientConfig parameters in the app config to give the SDK a choice as to what they want to use. I could also easily add a `realm_app_config_get_sync_client_config()` function that returns a copy of the sync_client_config, but I thought that might be confusing/misleading...

Reply:
@ironage - since the `realm_sync_client_config_new()` is not needed if REALM_APP_SERVICES is enabled, I just updated the `realm_sync_client_config_t` to be one of the two different types depending on the REALM_APP_SERVICES value:
* If it is set, `realm_app_config_get_sync_client_config()` is now available to get the sync client config from app config.
* If it is not set, `realm_sync_client_config_new()` is available to create the sync client config to use with the `realm_sync_manager_create()`, which is also only available if REALM_APP_SERVICES is not set.